### PR TITLE
Add ReviewPage and refactor a bit

### PR DESCRIPTION
**Pull request by Thrillberg**
## Summary
This PR creates the second page in the form upload flow, the `ReviewPage`. It also includes some refactors. The functionality is taken out of this draft PR (which I will close once I've merged it all in pieces): https://github.com/department-of-veterans-affairs/vets-website/pull/29569

This PR has some TODOs around displaying SSN and other PII, captured in this other ticket: https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team/83019

## Related issue(s)
https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/1257

### CODE REVIEW

File: src/applications/simple-forms/form-upload/containers/FormUploadApp.jsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import PropTypes from 'prop-types';
 
 import { RequiredLoginView } from '@department-of-veterans-affairs/platform-user/RequiredLoginView';
 
-function FormUploadApp({ user }) {
+const FormUploadApp = ({ user }) => {
   useEffect(() => {
     document.title = 'Form Upload | Veterans Affairs';
   }, []);
```
Comment (pennja) [inline src/applications/simple-forms/form-upload/containers/FormUploadApp.jsx:11]: While you're refactoring, I think you can get rid of the `connect` logic all together via `useSelector`:

```js
import React, { useEffect } from 'react';
import { useSelector } from 'react-redux'; 
import { Outlet } from 'react-router-dom-v5-compat';
import PropTypes from 'prop-types';

import { RequiredLoginView } from '@department-of-veterans-affairs/platform-user/RequiredLoginView';

const FormUploadApp = ({ user }) => {
  const user = useSelector(state => state.user);
   
  useEffect(() => {
    document.title = 'Form Upload | Veterans Affairs';
  }, []);

  return (
    <RequiredLoginView user={user} serviceRequired={[]}>
      <Outlet />
    </RequiredLoginView>
  );
};

FormUploadApp.propTypes = {
  user: PropTypes.object,
};

export default FormUploadApp;
```

**Review by pennja — DISMISSED** [review]
Looks good, great work! 👍

**Comment by Thrillberg** [comment]
I've updated the styling to show the blue left border.

Screenshot 2024-05-14 at 11 18 49 AM

**Review by pennja — APPROVED** [review]
Looks good, great work! 👍

**Pull request merged**